discord/poll.py

```diff
@@ -0,0 +1,624 @@
+"""
+The MIT License (MIT)
+
+Copyright (c) 2015-present Rapptz
+
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the "Software"),
+to deal in the Software without restriction, including without limitation
+the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+"""
+
+from __future__ import annotations
+
+
+from typing import Optional, List, TYPE_CHECKING, Union, AsyncIterator
+
+import datetime
+
+from .enums import PollLayoutType, try_enum
+from . import utils
+from .emoji import PartialEmoji, Emoji
+from .user import User
+from .object import Object
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
+
+    from .message import Message
+    from .abc import Snowflake
+    from .state import ConnectionState
+    from .member import Member
+
+    from .types.poll import (
+        PollCreate as PollCreatePayload,
+        PollMedia as PollMediaPayload,
+        PollAnswerCount as PollAnswerCountPayload,
+        Poll as PollPayload,
+        PollAnswerWithID as PollAnswerWithIDPayload,
+        PollResult as PollResultPayload,
+        PollAnswer as PollAnswerPayload,
+    )
+
+
+__all__ = (
+    'Poll',
+    'PollAnswer',
+    'PollMedia',
+)
+
+MISSING = utils.MISSING
+PollMediaEmoji = Union[PartialEmoji, Emoji, str]
+
+
+class PollMedia:
```

Comment:
This isn't a `NamedTuple`, nor should it be. Either way, this is missing `versionadded`.